### PR TITLE
feat(roles): Return access/scopes on Teams

### PR DESCRIPTION
**Pull request by leedongwei**

### CODE REVIEW

File: tests/sentry/api/serializers/test_team.py [hunk 1]
```diff
@@ -67,41 +73,47 @@ def test_member_count_does_not_include_invite_requests(self):
         result = serialize(team, self.user)
         assert result["memberCount"] == 2
 
-    def test_member_access(self):
+    def test_member(self):
```
Comment (nhsiehgit) [inline tests/sentry/api/serializers/test_team.py:76]: 🤔 

can we make this more descriptive of what we're testing on the team member?

File: tests/sentry/api/serializers/test_team.py [hunk 2]
```diff
@@ -110,18 +122,50 @@ def test_admin_access(self):
...
-        self.create_team_membership(user=user, team=team)
+        self.create_team_membership(user=user, team=team, role="admin")
+        result = serialize(team, user)
+        # after giving them access to team
+        assert result["access"] == TEAM_ADMIN["scopes"]
+        assert result["hasAccess"] is True
+        assert result["isMember"] is True
+        assert result["teamRole"] == TEAM_ADMIN["id"]
+
+    def test_admin(self):
```
Comment (nhsiehgit) [inline tests/sentry/api/serializers/test_team.py:138]: more descriptive test names

**Review by nhsiehgit — COMMENTED** [review]
nits:

personally for tests, i'd prefer if each test specifically called out what we wanted to test.
even if this means breaking them up into a bunch of smaller tests.

makes it much easier to understand what we're trying to achieve, and decouples the tests 🤷

File: src/sentry/api/serializers/models/team.py [hunk 3]
```diff
@@ -224,22 +224,10 @@ def get_attrs(
...
-                if top_org_role is not None:
-                    minimum_team_role = roles.get_minimum_team_role(top_org_role)
-                    if (
-                        not team_role
-                        or minimum_team_role.priority
-                        > team_roles.get(team_role.lower().replace("team ", "")).priority
-                    ):
-                        team_role = minimum_team_role.id
+            org_roles = all_org_roles.get(team.organization_id) or []
+            team_role_id = team_memberships.get(team.id)
```
[the remark this answers is not in the document]
Comment (leedongwei) [reply src/sentry/api/serializers/models/team.py:229]: They call it `id` here: https://github.com/getsentry/sentry/blob/1e58a48ffa1d550612f30575b331163c6ba34794/src/sentry/conf/server.py#L2002

**Review by nhsiehgit — COMMENTED** [review]
generally looks alright, but i feel i don't have enough context to properly give a sign off.

Will let @cathteng have the final say 😅

File: tests/sentry/api/serializers/test_team.py [hunk 4]
```diff
@@ -1,35 +1,50 @@
...
         }
 
+    def test_simple_with_group_org_role(self):
+        user = self.create_user(username="foo")
+        organization = self.create_organization()
+        team = self.create_team(organization=organization, org_role="manager")
+        self.create_member(user=user, organization=organization, role="owner")
+
+        result = serialize(team, user)
+        assert result["orgRole"] == "manager"
```
Comment (cathteng) [inline tests/sentry/api/serializers/test_team.py:46]: ah this needs to be renamed

**Review by cathteng — APPROVED** [review]
lgtm

**Pull request merged**